docker/docker_image_dest.go

```diff
@@ -213,17 +214,22 @@ func (d *dockerImageDestination) ReapplyBlob(info types.BlobInfo) (types.BlobInf
 // FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
 // If the destination is in principle available, refuses this manifest type (e.g. it does not recognize the schema),
 // but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
-func (d *dockerImageDestination) PutManifest(m []byte) error {
+func (d *dockerImageDestination) PutManifest(m []byte, instanceDigest *digest.Digest) error {
 	digest, err := manifest.Digest(m)
 	if err != nil {
 		return err
 	}
-	d.manifestDigest = digest
-
 	refTail, err := d.ref.tagOrDigest()
 	if err != nil {
 		return err
 	}
+	if instanceDigest != nil {
+		digest = *instanceDigest
+		refTail = instanceDigest.String()
+	} else {
+		d.manifestDigest = digest
```

Comment:
(Both the `digest` and `refTail` computations can be moved inside this `else`.)

Reply:
Makes sense.  Fixed.